isminet/clients/base.py

```
@@ -1,168 +1,211 @@
+                    return response_model(**data["data"][0])
+                elif "data" in data and not data["data"]:
+                    # If data is empty list, return empty model
+                    return response_model()
+                else:
+                    # Otherwise, try to parse entire response
+                    return response_model(**data)
+            return data
+        except (ConnectionError, Timeout):
+            raise
+        except ValidationError as e:
+            raise ResponseValidationError(f"Response validation failed: {str(e)}", e)
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)